### PR TITLE
Refactor Methods

### PR DESCRIPTION
**Pull request by takehiromouri**

### CODE REVIEW

File: app/controllers/bid_acceptances_controller.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 class BidAcceptancesController < ApplicationController
+	before_action :authenticate_broker_or_forward_freight!
```
Comment (takehiromouri) [inline app/controllers/bid_acceptances_controller.rb]: @rabingaire I added this method, we need to authenticate if there's a user, even if there's no visible user interface for it :)

File: app/models/job.rb [hunk 2]
```diff
@@ -9,20 +8,13 @@ class Job < ApplicationRecord
...
 
-	has_many :bids
-	
-
-	# def clone_job
-	# 	job_type = job_type.broker? forward_freight : broker
-	# 	job = Job.new(business_user_id: current_business_user.id, job_type: job_type, agent_id: nil)
-	# 	job.save
-	# end
+	has_many :bids
```
Comment (takehiromouri) [inline app/models/job.rb]: Reorganized and grouped similar things together here

**Comment by takehiromouri** [comment]
Some minor refactoring here.

Quick question @aadeshere1, in this method:

```
def authenticate_any!
    if broker_signed_in?
        authenticate_broker!
    elsif forward_freight_signed_in?
        authenticate_forward_freight!
    elsif business_user_signed_in?
        authenticate_business_user!
    else
        flash[:alert] = "Please sign in."
        redirect_to root_path
    end
  end
```

I'm wondering how this might work - could we do something like this instead:

```
def authenticate_any!
  return if broker_signed_in? || forward_freight_signed_in? || business_user_signed_in?
  flash[:alert] = "Please sign in."
  redirect_to root_path
end
```

I'm thinking that this might be redundant:

```
if broker_signed_in?
        authenticate_broker!
```

If the broker isn't signed in, it's never going to go to `authenticate_broker!`, it's just going to go to the next `elsif` if I'm not mistaken :)

Let me know your thoughts on this, and maybe we should write some tests for this even though it's a protected method :)

**Comment by aadeshere1** [comment]
@takehiromouri  Initially the code was

```
def authenticate_any!
    if broker_signed_in?
        true
    elsif forward_freight_signed_in?
        true
    else
        authenticate_business_user!
    end
  end
```

which is similar to your approach. :) Let's change it as you suggested :)

**Comment by takehiromouri** [comment]
Awesome :) Could you push this and maybe write some tests for it as well? :D

**Comment by aadeshere1** [comment]
@takehiromouri I guess we can merge this branch :)

**Pull request merged**